influxdb3_id/src/serialize.rs

```diff
@@ -0,0 +1,207 @@
+use std::{
+    marker::PhantomData,
+    ops::{Deref, DerefMut},
+};
+
+use hashbrown::{
+    hash_map::{IntoIter, Iter, IterMut},
+    HashMap,
+};
+use serde::{
+    de::{self, SeqAccess, Visitor},
+    ser::SerializeSeq,
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+
+/// A new-type around a `HashMap` that provides special serialization and deserialization behaviour.
+///
+/// Specifically, it will be serialized as a vector of tuples, each tuple containing a key-value
+/// pair from the map. Deserialization assumes said serialization, and deserializes from the vector
+/// of tuples back into the map. Traits like `Deref`, `From`, etc. are implemented on this type such
+/// that it can be used as a `HashMap`.
+///
+/// During deserialization, there are no duplicate keys allowed. If duplicates are found, an error
+/// will be thrown.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SerdeVecHashMap<K: Eq + std::hash::Hash, V>(HashMap<K, V>);
```

Comment:
If this is mainly used for ID maps, which are integers, we might be able to use FNV hash: https://doc.servo.org/fnv/

Reply:
I opened https://github.com/influxdata/influxdb/issues/25494